embuilder.py

```diff
@@ -191,6 +132,8 @@ def main():
   def is_flag(arg):
     return arg.startswith('--')
 
+  shared.check_sanity()
```

Comment:
i vaguely remember this was here for a reason - that we must check here so the cache is invalidated up front, rather than later? please add a comment explaining this. also, I think this should run even without main - how about putting it at the top level at the bottom of the file before checking if we are in a main program?